src/components/Components/OpenSimGUIScene.tsx

```diff
@@ -248,75 +250,50 @@ const OpenSimGUIScene: React.FC<OpenSimSceneProps> = ({ currentModelPath, suppor
           // Start new animation
           const action = mixers[viewerState.currentAnimationIndex]?.clipAction(viewerState.animations[viewerState.currentAnimationIndex]);
           action?.play();
-
-          // Update animation time immediately when animation changes
-          if (action) {
-            const currentTime = action.time;
-            viewerState.setCurrentAnimationTime(currentTime);
-          }
         }
 
-        // Handle animation playback with the three cases
-        const idx = viewerState.currentAnimationIndex;
+        // Handle animation playback
         if (idx !== -1) {
           const mixer = mixers[idx];
           const action = mixer.clipAction(viewerState.animations[idx]);
           const duration = action.getClip().duration;
 
-          // CASE 1: PLAYING (animating OR GUI animating)
+          // If we're animating (playing), update the mixer with delta time
           if (viewerState.animating || curState.isGUIAnimating) {
-            // Handle slider-driven time when animating
-            if (curState.currentFrame !== startTime && viewerState.animating) {
-              const framePercentage = curState.currentFrame / 100;
-              const currentTimeInSlider = duration * framePercentage;
-              action.time = currentTimeInSlider;
-            }
-
-            mixer.update(delta * viewerState.animationSpeed);
+            if (curState.isGuiMode)
+              mixer.update(delta * modelUIState.guiAnimationSpeed);
```

Comment:
If we can set the time here rather than update that would be betterso the motion and dolly are sync'ed